File: src/main/java/com/timepath/plaf/OS.java
package com.timepath.plaf;

import org.jetbrains.annotations.NotNull;

import java.util.logging.Level;
import java.util.logging.Logger;

/**
 * @author TimePath
 */
public enum OS {
    Windows,
    OSX,
    Linux,
    Other;
    private static final Logger LOG = Logger.getLogger(OS.class.getName());
    @NotNull
    private static final OS system;

    static {
        @NotNull String osVer = System.getProperty("os.name").toLowerCase();
        if (osVer.contains("windows")) {
            system = Windows;
        } else if (osVer.contains("mac os x") || osVer.contains("OS X") || osVer.contains("mac")) {
            system = OSX;
        } else if (osVer.contains("Linux") || osVer.contains("nix") || osVer.contains("nux")) {
            system = Linux;
        } else {
            system = Other;
            LOG.log(Level.INFO, "OS string: {0}", osVer);
        }
        LOG.log(Level.INFO, "OS: {0}", system);
    }

    @NotNull
    public static OS get() {
        return system;
    }

    public static boolean isWindows() {
        return system == Windows;
    }

    public static boolean isMac() {
        return system == OSX;
    }

    public static boolean isLinux() {
        return system == Linux;
    }
}

File: src/main/kotlin/com/timepath/plaf/OS.kt
package com.timepath.plaf


import com.timepath.Logger
import java.util.logging.Level

/**
 * @author TimePath
 */
public enum class OS {
    Windows,
    OSX,
    Linux,
    Other;

    companion object {

        private val LOG = Logger()

        public fun get(): OS {
            val osVer = System.getProperty("os.name").toLowerCase()
            val system = when {
                "windows" in osVer -> Windows
                "mac os x" in osVer, "OS X" in osVer, "mac" in osVer -> OSX
                "Linux" in osVer, "nix" in osVer, "nux" in osVer -> Linux
                else -> {
                    LOG.log(Level.WARNING) { "Unknown OS string: $osVer" }
                    Other
                }
            }
            LOG.log(Level.INFO) { "OS: $system" }
            return system
        }

        public fun isWindows(): Boolean = get() == Windows

        public fun isMac(): Boolean = get() == OSX

        public fun isLinux(): Boolean = get() == Linux
    }

}
